src/loader.c

```diff
@@ -29,10 +29,38 @@
 #include <dlfcn.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 
 static void* s_impl_library = 0;
 static struct wpe_loader_interface* s_impl_loader = 0;
 
+#ifndef WPE_BACKEND
+#define IMPL_LIBRARY_NAME_BUFFER_SIZE 512
+static char* s_impl_library_name;
+static char s_impl_library_name_buffer[IMPL_LIBRARY_NAME_BUFFER_SIZE];
+#endif
+
+#ifndef WPE_BACKEND
+static void
+wpe_loader_set_impl_library_name(const char* impl_library_name)
+{
+    size_t len;
+
+    if (!impl_library_name)
+        return;
+
+    len = strlen(impl_library_name) + 1;
+    if (len == 1)
+        return;
+
+    if (len > IMPL_LIBRARY_NAME_BUFFER_SIZE)
+        s_impl_library_name = (char *)malloc(len);
```

Comment:
If `wpe_loader_set_impl_library_name()` is called more than once with strings longer than `IMPL_LIBRARY_NAME_BUFFER_SIZE` memory will be leaked.

The simplest solutions I can think of are:

- Make the static buffer have size `PATH_MAX` and never allocate memory dynamically.
- Check whether `s_impl_library_name` points to `s_impl_library_name_buffer`, and if not, always `free()` before allocating memory again.

The second option might be desirable because `PATH_MAX` can be “big” e.g. in modern GNU/Linux it's 4 KiB and that would be a little bit of a waste of memory.

Reply:
set_impl_library_name() is private and can't be called multiple times, it's protected by s_impl_library. I don't want to use PATH_MAX in any case, it has different values in different platforms, and it's not a real limit at all, since it actually depends on the file system.

Reply:
Ah, I missed the part where it's only called once, you're right. Maybe adding a comment here mentioning that the function can't be called multiple times will be helpful for future reference.

I also think that it's better to allocate the size needed, instead of using `PATH_MAX` or its most correct variant, `pathconf(..., _PC_PATH_MAX)`, so let's leave it as-is.